### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by tajemouti**
## Hi Reviewer 👋 ##
Below is a list of the accomplished tasks while working on this milestone:

1. Created a feature branch for this milestone.
2. Created a class called Nameable.
3. Implemented a method correct_name that raises a NotImplementedError.
4. Modified the Person class to inherit from the Nameable class.
5. Implemented the correct_name method in the Person class to return the name attribute.
6. Created a class called BaseDecorator.
7. Assigned a nameable object from the parameters to an instance variable.
8. Implemented the correct_name method in the BaseDecorator class.
9. Created a class called CapitalizeDecorator that inherits from the BaseDecorator.
10. Implemented the correct_name method in the CapitalizeDecorator class.
11. Created a class called TrimmerDecorator that inherits from the BaseDecorator.
12. Implemented the correct_name method in the TrimmerDecorator class.
13. Created a Person object,  and verified the capitalized and trimmed name.

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
+  end
+end
```
Comment (Reem-lab) [inline trimmer_decorator.rb]: - [ ] Great job working on the decorator class but here in the trimmer class you have something to add a condition and to make sure  that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word. as mentioned in the requirements 🔥🤩💯🚀 

- So you should check first it is longer than 10 then trim the word ✅
image

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @tajemouti, 👋🏻👋🏻

Good job so far! 🔥💯✅✨
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ]  Kudos for working on the decorator class🚀
- [ ]  No Linters error ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 1 above)
Comment (tajemouti) [reply trimmer_decorator.rb]: Hi @Reem-lab 
Thank you for your review and feedback. I appreciate your attention to detail. I understand your point, and I've carefully reviewed the code.
The code indeed fulfills the requirement of truncating the name to a maximum of 10 characters when it exceeds that length. It does so by using the slice method.
However, I acknowledge your feedback, and I'm willing to make the suggested change, even though the code is already functioning correctly in this regard. Your feedback is valuable, and I want to ensure that the code aligns with the intent of the requirements.
I will update the code as per your suggestion to make it even more explicit. Thanks again for the review, and I appreciate your help in improving the code.
Cheers

**Review by shahadat3669 — APPROVED** [review]
Hi @tajemouti, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅  the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.
✅ CapitalizeDecorator working fine.
✅ The code is top-notch! 💯

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**